collect_app/src/test/java/org/odk/collect/android/application/initialization/SavepointsImporterTest.kt

```diff
@@ -0,0 +1,305 @@
+package org.odk.collect.android.application.initialization
+
+import android.app.Application
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.contains
+import org.hamcrest.Matchers.equalTo
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.odk.collect.android.injection.DaggerUtils
+import org.odk.collect.android.storage.StorageSubdirectory
+import org.odk.collect.forms.Form
+import org.odk.collect.forms.instances.Instance
+import org.odk.collect.forms.savepoints.Savepoint
+import org.odk.collect.formstest.FormFixtures
+import org.odk.collect.formstest.InstanceFixtures
+import org.odk.collect.projects.Project
+import org.odk.collect.shared.TimeInMs
+import org.odk.collect.shared.strings.RandomString
+import java.io.File
+
+@RunWith(AndroidJUnit4::class)
+class SavepointsImporterTest {
+    private val component = DaggerUtils.getComponent(ApplicationProvider.getApplicationContext<Context>() as Application)
+
+    private val projectsRepository = component.projectsRepository()
+    private val projectDependencyProviderFactory = component.projectDependencyProviderFactory()
+
+    private val savepointsImporter =
+        SavepointsImporter(projectsRepository, projectDependencyProviderFactory)
+
+    private val project = projectsRepository.save(Project.DEMO_PROJECT)
+    private val projectDependencyProvider = projectDependencyProviderFactory.create(project.uuid)
+
+    @Test
+    fun ifABlankFormHasNoSavepoint_nothingShouldBeImported() {
+        // create blank forms
+        createBlankForm(project, "sampleForm", "1")
+
+        // trigger importing
+        savepointsImporter.run()
+
+        // verify import
+        val savepoints = projectDependencyProvider.savepointsRepository.getAll()
+        assertThat(savepoints.isEmpty(), equalTo(true))
+    }
+
+    @Test
+    fun ifABlankFormHasASavepointCreatedEarlierThanTheForm_nothingShouldBeImported() {
+        // create blank forms
+        createBlankForm(project, "sampleForm", "1", date = System.currentTimeMillis() + TimeInMs.ONE_HOUR)
```

Comment:
I'm guessing we can't used fix times here because of we call the real `lastModified` on the file right?

Reply:
Yes.